### PR TITLE
FireMessage

### PR DESCRIPTION
**Pull request by swalker2m**
This is the first of a couple of PRs related to sending execution events and associated data to the Fast Image Reduction Engine (?), or FIRE system.  At this point the messages are created in the appropriate place in the WDBA (Web-Database Access) bundle but not yet posted.  The bulk of the work is WDBA refactoring to enable a simple test case.

Example `FireMessage` containing all the information in the requirements doc:

```
{
  "too" : "None",
  "observationId" : "GS-2021B-Q-9999-1",
  "fileNames" : [
    "S20210813S0001.fits"
  ],
  "id" : "64d3be13-351b-47e2-bb70-db1cddf92006",
  "visitStartTime" : "2021-08-13T14:11:51.711Z",
  "sequence" : {
    "completedDatasetCount" : 1,
    "totalStepCount" : 3,
    "durationMilliseconds" : 345000,
    "completedStepCount" : 1,
    "totalDatasetCount" : 3
  },
  "programId" : "GS-2021B-Q-9999",
  "eventTime" : "2021-08-13T14:11:51.712Z",
  "nature" : "Start Dataset"
}
```

### CODE REVIEW

File: bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/server/osgi/WDBAServerActivator.java [hunk 1]
```diff
@@ -79,9 +87,36 @@ public WdbaContext addingService(ServiceReference<IDBDatabaseService> ref) {
                         }
 
                         final WdbaContext ctx = new WdbaContext(site, new WdbaGlueService(db, user), user);
+
+                        fireService     = FireService.loggingOnly(db);
```
Comment (swalker2m) [inline bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/server/osgi/WDBAServerActivator.java:91]: For now, it only logs the `FireMessage`s as the are produced.  They will need to be posted to a server in production.

File: bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/session/DBUpdateService.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
+//
+// $Id: DBUpdateService.java 887 2007-07-04 15:38:49Z gillies $
```
Comment (swalker2m) [inline bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/session/DBUpdateService.java:2]: There are some changes but it was mostly just moved from another package.

File: bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/session/SessionXmlRpcHandler.java [hunk 3]
```diff
@@ -72,8 +74,11 @@ private static void log(String method, String... args) {
     // this has to be set from the activator and then later used by the
     // SessionXmlRpcHandler that is eventually created.
     private static SessionManagement sm = null;
-    public static synchronized void setContext(WdbaContext context) {
-        sm = (context == null) ? null : new SessionManagement(context, new ProductionSessionConfiguration(context));
+    public static synchronized void setContext(
+        WdbaContext context,
+        Consumer<ExecEvent> eventConsumer
```
Comment (swalker2m) [inline bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/session/SessionXmlRpcHandler.java:79]: The `eventConsumer` is configured depending upon whether we're in production or testing.  It receives an event and does something appropriate with it.

Comment (swalker2m) [inline bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/session/SessionXmlRpcHandler.java:79]: The `eventConsumer` is configured depending upon whether we're in production or testing.  It receives an event and does something appropriate with it.

File: bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/session/SessionManagement.java [hunk 4]
```diff
@@ -33,17 +36,17 @@
 
     private int _sessionCount = 0;
     private final WdbaContext _ctx;
-    private ISessionConfiguration _config;
-    private final Map<String, Session> _sessions = new HashMap<String, Session>();
+    private final Map<String, Session> _sessions = new HashMap<>();
```
Comment (cquiroz) [inline bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/session/SessionManagement.java:39]: Didn't know WDBA had sessions

**Review by cquiroz — APPROVED** [review]
I can't really comment on the details but LGTM

**Pull request merged**